ethsigner/signer/file-based/src/test/java/tech/pegasys/ethsigner/signer/filebased/FileBasedTransactionSignerTest.java

```diff
@@ -47,60 +47,42 @@ public static void createKeyFile() {
 
   @Test
   public void success() throws IOException {
-    final FileBasedSignerConfig configMock = mock(FileBasedSignerConfig.class);
-
     final File keyFile = new File(fileName);
     final File pwdFile = createFile(MY_PASSWORD);
 
-    when(configMock.getPasswordFilePath()).thenReturn(pwdFile.toPath());
-    when(configMock.getKeyPath()).thenReturn(keyFile.toPath());
-
-    final TransactionSigner signer = new FileBasedSignerBuilder(configMock).build();
+    final TransactionSigner signer =
+        FileBasedSignerFactory.createSigner(keyFile.toPath(), pwdFile.toPath());
 
     assertThat(signer).isNotNull();
+    assertThat(signer.getAddress()).isNotEmpty();
   }
 
-  @Test
+  @Test(expected = TransactionSignerInitializationException.class)
```

Comment:
The pattern used in Pantheon seems to be one where the exception is both caught and interrogated for class and state (message / key / enum)

e.g.
```
    final Throwable thrown = catchThrowable(() -> ileBasedSignerFactory.createSigner(keyFile.toPath(), pwdFile.toPath());

    assertThat(thrown)
        .isInstanceOf(TransactionSignerInitializationException.class)
        .hasMessage("Whatever message is expected");
```

Is there no reason to interrogate the exception beyond the class?

Reply:
Yup - will change it to the Pantheon way.